python/cudf/cudf/_lib/text.pyx

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2020-2021, NVIDIA CORPORATION.
+
+import cudf
+
+from cython.operator cimport dereference
+from libcpp.memory cimport make_unique, unique_ptr
+from libcpp.string cimport string
+from libcpp.utility cimport move
+
+from cudf._lib.column cimport Column
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.cpp.io.text cimport (
+    data_chunk_source,
+    make_source,
+    make_source_from_file,
+    multibyte_split,
+)
+
+
+cpdef read_text(object filepaths_or_buffers,
+                object delimiter=None):
+    """
+    Cython function to call into libcudf API, see `read_text`.
+
+    See Also
+    --------
+    cudf.io.text.read_text
+    """
+    cdef string filename = filepaths_or_buffers.encode()
+    cdef string delim = delimiter.encode()
+
+    cdef unique_ptr[data_chunk_source] datasource
+    cdef unique_ptr[column] c_col
+
+    with nogil:
+        datasource = move(make_source_from_file(filename))
+        c_col = move(multibyte_split(dereference(datasource), delim))
+
+    col = Column.from_unique_ptr(move(c_col))
+    df = cudf.DataFrame._from_data(
+        cudf.core.column_accessor.ColumnAccessor({"col_name": col}))
+
+    return df
```

Comment:
No need to return a `DataFrame` here, we're typically just returning a dict of columns from Cython APIs now.

```suggestion
    return {None: col}
```